### DISCUSSION TITLE
GSOC 2024 - Making ZeroMQ a first-class feature of concore

### DISCUSSION THREAD

**Discussion by pratik-choudhari** [ControlCore-Project/concore#113] [General]
Hi @pradeeban ,
As part of GSOC 2024, I am highly interested in applying for implementation of messages queues in concore. Past 2 days, I have spent my time studying the source code and bash files of this project, installing the system and executing demo files on my local machine. Thus, I now have a clear idea of how the current system works. I have a few interesting modifications in mind for this project. Really excited to join this project!

**Comment by pratik-choudhari** [comment]
@pradeeban How can we submit a draft proposal? Do we have to email it?

**Reply by pradeeban** [reply]
Yes, you can email it.

**Comment by harshal-rembhotkar** [comment]
Hello @pradeeban, i want to dive deep into Concore-library. Can you suggest some resources about Concore-library ?

**Reply by pradeeban** [reply]
The documentation is a good place to start - https://control-core.readthedocs.io/en/latest/

**Reply by harshal-rembhotkar** [reply]
Thank you.

**Comment by harshal-rembhotkar** [comment]
Hello @pradeeban, I'm curious to know, what Java-specific tools would you recommend for implementing the Concore library, similar to how CMake and Conan are utilized in the C++ ecosystem? Specifically, I'm interested in suggestions for dependency management, build automation, and project structuring tools like Maven and Gradle or any others. Any insights or recommendations would be highly valued!

**Reply by pradeeban** [reply]
Yes. Apache Maven works well for project management in Java.
